src/main/java/com/codeborne/selenide/appium/commands/AppiumScrollTo.java

```diff
@@ -0,0 +1,78 @@
+package com.codeborne.selenide.appium.commands;
+
+import com.codeborne.selenide.SelenideElement;
+import com.codeborne.selenide.commands.ScrollTo;
+import com.codeborne.selenide.impl.WebElementSource;
+import io.appium.java_client.AppiumDriver;
+import org.openqa.selenium.Dimension;
+import org.openqa.selenium.NoSuchElementException;
+import org.openqa.selenium.interactions.PointerInput;
+import org.openqa.selenium.interactions.Sequence;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import javax.annotation.ParametersAreNonnullByDefault;
+import java.util.Optional;
+
+import static com.codeborne.selenide.appium.WebdriverUnwrapper.cast;
+import static java.time.Duration.ofMillis;
+import static java.util.Collections.singletonList;
+
+
+@ParametersAreNonnullByDefault
+public class AppiumScrollTo extends ScrollTo {
+
+  private AppiumDriver appiumDriver;
+
+  @Override
+  @Nonnull
+  public SelenideElement execute(SelenideElement proxy, WebElementSource locator, @Nullable Object[] args) {
+    Optional<AppiumDriver> appiumDriverOptional = cast(locator.driver(), AppiumDriver.class);
+    if (!appiumDriverOptional.isPresent()) {
+      return super.execute(proxy, locator, args);
+    }
+
+    this.appiumDriver = appiumDriverOptional.get();
+
+    String previousPageSource = "";
+    while (isElementNotDisplayed(locator) && isNotEndOfPage(previousPageSource)) {
```

Comment:
Add time\count condition to break potentially endless loop. There are cases where `isNotEndOfPage(previousPageSource)` will be true forever(e.g. facebook news feed)

Reply:
@BorisOsipov - Could you please suggest a timeout or count of scrolls. Should I take it as 1 min of timeout or 30 scroll units?

Reply:
@BorisOsipov - For now I have added a max wait of 1 min. Please see if this implementation is okay. Otherwise, I can create a class called ScrollOptions with timeout and scroll directions.

Reply:
>Should I take it as 1 min of timeout or 30 scroll units?

@amuthansakthivel let's [copy approach from android uiautomator](https://android.googlesource.com/platform/frameworks/testing/+/f612e6a05f47b28ae0f5715545658c08dd759dd7/uiautomator/library/src/com/android/uiautomator/core/UiScrollable.java#38) and use `mMaxSearchSwipes=30`.

Reply:
@BorisOsipov - I have already implemented with stopwatch class and timeout of 1 min. (For 1 min, approximately it takes 48 swipes counts)

30 swipe counts will be approximately 7 times the screen size. (I scroll 1/4th of screen size every time)

Please confirm.

Reply:
@amuthansakthivel I am a bit affraid of stopwatch because scroll\getPageSource actions can be executed slowly in cloud device providers like SauseLab\BrowserStack. So It will work for 1 min in local env, but fails in cloud\grid.  Max swipe counts don't have such edge cases. What do you think? Or maybe my consern groundless?

Reply:
@BorisOsipov - You are correct. We cannot guarantee how many swipes would happen in 1 min. I have refactored the code now to scroll max upto 30 swipes.